test/cypress/integration/group-proposals.spec.js

```diff
@@ -236,33 +236,54 @@ describe('Proposals - Add members', () => {
       })
   })
 
-  it('It should display that no one is doing any contribution', () => {
+  it('User click on contribution page and see the card', () => {
     cy.getByDT('contributionsLink').click()
     cy.getByDT('addIncomeDetailsFirstCard').should('contain', 'Add your income details')
+  })
+
+  it('Open the income detail modal should ask only if you want to receive or pledge', () => {
     cy.getByDT('openIncomeDetailModal').click()
     cy.getByDT('introIncomeOrPledge').should('not.be.visible')
+  })
+
+  it('After selecting the pledge option, it should ask you how much you want to pledge', () => {
     cy.getByDT('dontNeedsIncomeRadio').click()
     cy.getByDT('introIncomeOrPledge').should('contain', 'How much do you want to pledge?')
+  })
+
+  it('After selecting the amount and close the modal it should show that no one is in need', () => {
```

Comment:
Good tests descriptions are hard to do and even harder to explain how to do them... The way I think about it is to write pragmatic test descriptions and focus on the final outcome of a task. For example, in this case, I would merge these 3 tests (244, 254 and 249), to reflect a user's "single real task".

> 'user1 fills their Income Details saying they are pledging 500$'.

That's the main goal in this chunk of tests. But keep the assertions for "edge cases" is totally fine (and desirable!), so you can keep the line asserting that introIncomeOrPledge doesn't exist with an inline comment to help other developers understand why we are "not.asserting" something.
```
cy.getByDT('introIncomeOrPledge').should('not.be.visible') // Make sure only the first question is visible
```